### PR TITLE
Create a new method to generalize the offensive ability usages

### PR DESCRIPTION
**Pull request by JasonKaz**

### CODE REVIEW

File: autoPlay.js [hunk 1]
```diff
@@ -297,37 +297,16 @@ function useClusterBombIfRelevant() {
...
+}
 
 function useNapalmIfRelevant() {
-	//Check if Napalm is purchased and cooled down
-	if (hasPurchasedAbility(ABILITIES.NAPALM)) {
-		if (isAbilityCoolingDown(ABILITIES.NAPALM)) {
-			return;
-		}
-		
-		//Check lane has monsters to burn
```
Comment (mouseas) [inline autoPlay.js:312]: While a generalized method would be helpful, you can't but out the logic in each of the methods.

Also, I think naming the method `useOffensiveAbilityIfAvailable` would be better, then just do the ability-specific logic before calling it, and you should be good.

**Comment by JasonKaz** [comment]
So with these new changes it allows us to have a configurable list of conditions for using abilities. I didn't move the medic or charm abilities because they seemed very specific and weren't easy to generalize.

**Comment by mouseas** [comment]
I need ~~2 people~~ one more person to independently test this for errors and functionality. (Thanks @furtivespy)

**Comment by furtivespy** [comment]
I haven't tested, but I've reviewed the code and the logic looks sound. Here are my questions/comments:
- Get rid of the changed useAbilityIfRelevant() functions. just call the generic from main.
- Lucky charms can be done - just null on all except enemy count 2+ (it's only not used on boss).
- Should abilityConditions be part of a global config item? Either way should probably stick with the standard of ALL_CAPS for global variables.

**Comment by JasonKaz** [comment]
All of @furtivespy 's changes were made, plus others.

We really should get these changes in sooner rather than later. It makes adding new ability/item using very trivial.

**Comment by JasonKaz** [comment]
We really should get these changes in sooner rather than later. It makes adding new ability/item using very trivial.

File: autoPlay.js [hunk 2]
```diff
@@ -1,458 +1,510 @@
...
-	var lowHP = 0;
-	var lowLane = 0;
-	var lowTarget = 0;
-	var lowPercentageHP = 0;
-	
-	// determine which lane and enemy is the optimal target
-	var enemyTypePriority = [
-		ENEMY_TYPE.TREASURE, 
-		ENEMY_TYPE.BOSS, 
+    // We can overlook spawners if all spawners are 40% hp or higher and a creep is under 10% hp
```
Comment (mouseas) [inline autoPlay.js]: Please revert the whitespace changes. The code is formatted with tabs, and you've switched a bunch of it to spaces.

**Comment by JasonKaz** [comment]
Should be fully converted now.

**Comment by mouseas** [comment]
@JasonKaz Great, now we just need someone to test it.

**Comment by win0na** [comment]
When this is ready, inform me and I will fix the various merge conflicts introduced by the filename change.

**Comment by JasonKaz** [comment]
I don't suppose just me testing it is enough?

**Comment by win0na** [comment]
As indicated by this comment (https://github.com/mouseas/steamSummerMinigame/pull/42#issuecomment-111672330), it would be beneficial to have at least one more person testing it to ensure it works properly.

This macro explains it perfectly.

descarga1

**Comment by mouseas** [comment]
This PR is too old to merge in, and code duplication has already been reduced sufficiently for the areas in question. I'm closing this PR.

**Pull request closed without merging**